libs/goaccumulate/goaccumulate.go

```diff
@@ -0,0 +1,247 @@
+
+}
+
+func CreateToken(url string, symbol string, precision string, signer string) {
+	fmt.Println("Creating new token " + symbol)
+}
+
+// -------------------------------------------------------------------------------
+// Accounts
+// -------------------------------------------------------------------------------
+
+func GetAccount(url string) (string, error) {
+	var rez api2.QueryResponse
+	u, err := url2.Parse(url)
+	if err != nil {
+		return "", fmt.Errorf("%q is not a valid Accumulate URL: %v\n", url, err)
+	}
+
+	params := api2.UrlQuery{}
+	params.Url = u.String()
+
+	data, err := json.Marshal(&params)
+	if err != nil {
+		return "", err
+	}
+
+	if err := Client.Request(context.Background(), "query", json.RawMessage(data), &rez); err != nil {
+		return PrintJsonRpcError(err)
+	}
+
+	return PrintQueryResponseV2(&rez)
+}
+
+func QrAccount(s string) (string, error) {
+	u, err := url2.Parse(s)
+	if err != nil {
+		return "", fmt.Errorf("%q is not a valid Accumulate URL: %v\n", s, err)
+	}
+
+	b := bytes.NewBufferString("")
+	qrterminal.GenerateWithConfig(u.String(), qrterminal.Config{
+		Level:          qrterminal.M,
+		Writer:         b,
+		HalfBlocks:     true,
+		BlackChar:      qrterminal.BLACK_BLACK,
+		BlackWhiteChar: qrterminal.BLACK_WHITE,
+		WhiteChar:      qrterminal.WHITE_WHITE,
+		WhiteBlackChar: qrterminal.WHITE_BLACK,
+		QuietZone:      2,
+	})
+
+	r, err := ioutil.ReadAll(b)
+	return string(r), err
+}
+
+func CreateAccount(url string, args []string) (string, error) {
+	actor, err := url2.Parse(url)
+	if err != nil {
+		return "", err
+	}
+
+	args, si, privKey, err := prepareSigner(actor, args)
+	if len(args) < 3 {
+		return "", fmt.Errorf("insufficient number of command line arguments")
+	}
+
+	accountUrl, err := url2.Parse(args[0])
+	if err != nil {
+		return "", fmt.Errorf("invalid account url %s", args[0])
+	}
+	if actor.Authority != accountUrl.Authority {
+		return "", fmt.Errorf("account url to create (%s) doesn't match the authority adi (%s)", accountUrl.Authority, actor.Authority)
+	}
+	tok, err := url2.Parse(args[1])
+	if err != nil {
+		return "", fmt.Errorf("invalid token url")
+	}
+
+	var keybook string
+	if len(args) > 2 {
+		kbu, err := url2.Parse(args[2])
+		if err != nil {
+			return "", fmt.Errorf("invalid key book url")
+		}
+		keybook = kbu.String()
+	}
+
+	tokenJson, err := Get(tok.String())
+	if err != nil {
+		return "", err
+	}
+	token := protocol.TokenIssuer{}
+	err = json.Unmarshal([]byte(tokenJson), &token)
+	if err != nil {
+		return "", fmt.Errorf("invalid token type %v", err)
+	}
+
+	tac := &protocol.TokenAccountCreate{}
+	tac.Url = accountUrl.String()
+	tac.TokenUrl = tok.String()
+	tac.KeyBookUrl = keybook
+
+	jsonData, err := json.Marshal(tac)
+	if err != nil {
+		return "", err
+	}
+
+	binaryData, err := tac.MarshalBinary()
+	if err != nil {
+		return "", err
+	}
+
+	nonce := nonceFromTimeNow()
+	params, err := prepareGenTxV2(jsonData, binaryData, actor, si, privKey, nonce)
+	if err != nil {
```

Comment:
we will not always have access to the private key.  For example, when passing on to a hardware wallet for signing.  need a signing callback or something similar